runtimes/peregrine/src/runtime_apis.rs

```diff
@@ -9,45 +9,60 @@ use cumulus_primitives_core::CollationInfo;
 use frame_support::{
 	genesis_builder_helper::{build_config, create_default_config},
 	pallet_prelude::{TransactionSource, TransactionValidity},
-	traits::PalletInfoAccess,
+	traits::{
+		fungibles::{metadata::Inspect as MetadataInspect, Inspect},
+		PalletInfoAccess,
+	},
 	weights::Weight,
 };
-use pallet_transaction_payment::{FeeDetails, RuntimeDispatchInfo};
-use sp_api::impl_runtime_apis;
-use sp_core::OpaqueMetadata;
-use sp_inherents::{CheckInherentsResult, InherentData};
-use sp_metadata_ir::RuntimeApiMetadataIR;
-use sp_runtime::{
-	traits::{Block as BlockT, TryConvert},
-	ApplyExtrinsicResult, KeyTypeId,
-};
-use sp_std::vec::Vec;
-use sp_version::{ApisVec, RuntimeVersion};
-
 use kilt_runtime_api_did::RawDidLinkedInfo;
 use kilt_support::traits::ItemFilter;
 use pallet_asset_switch::xcm::AccountId32ToAccountId32JunctionConverter;
+use pallet_bonded_coins::{
+	curves::{
+		balance_to_fixed, fixed_to_balance, lmsr::LMSRParameters, polynomial::PolynomialParameters,
+		square_root::SquareRootParameters, BondingFunction, Curve,
+	},
+	PoolDetailsOf, Pools, Round,
+};
+use pallet_bonded_currency_runtime_api::{
+	BondedCurrencyDetails, CollateralDetails, HumanReadablePoolDetails, Operation,
+};
 use pallet_did_lookup::{linkable_account::LinkableAccountId, ConnectionRecord};
 use pallet_dip_provider::traits::IdentityProvider;
+use pallet_transaction_payment::{FeeDetails, RuntimeDispatchInfo};
 use pallet_web3_names::{web3_name::Web3NameOwnership, Web3NameOf};
+use parity_scale_codec::alloc::string::{String, ToString};
 use public_credentials::CredentialEntry;
 use runtime_common::{
 	asset_switch::runtime_api::Error as AssetSwitchApiError,
 	assets::{AssetDid, PublicCredentialsFilter},
 	authorization::AuthorizationId,
+	bonded_currencies::{self, runtime_api::Error as BondedCurrencyError, FixedPoint, FixedPointUnderlyingType},
```

Comment:
This is ugly as well 😄

Reply:
[baad1e3](https://github.com/KILTprotocol/kilt-node/pull/828/commits/baad1e3b5d31fbc465bca7f4e74ee6985a07a7a3)